pkg/controllers/federatedcluster/clusterjoin.go

```diff
@@ -93,6 +94,9 @@ func (c *FederatedClusterController) handleNotJoinedCluster(
 		time.Since(joinedCondition.LastTransitionTime.Time) > c.clusterJoinTimeout {
 		// Join timed out
 		logger.Error(nil, "Cluster join timed out")
+		c.metrics.Duration("cluster_joined_duration", cluster.CreationTimestamp.Time,
+			stats.Tag{Name: "cluster_name", Value: cluster.Name},
+			stats.Tag{Name: "result", Value: "error"})
```

Comment:
```suggestion
			stats.Tag{Name: "result", Value: "timeout"})
```
timeout or error? /cc @limhawjia what is your opinion？

Reply:
i realized that cluster joining may also fail if the member is already associated with another kubeadmiral control plane (see lines 155-170 below). should we record a metric for that case as well? @mrlihanbo

Reply:
in any case, since joining might fail due to various reasons, we should be more explicit. i prefer `Value: "timeout"`